lib/health-indicator/database/mikro-orm.health.ts

```diff
@@ -0,0 +1,141 @@
+   */
+  connection?: any;
+  /**
+   * The amount of time the check should require in ms
+   */
+  timeout?: number;
+}
+
+/**
+ * The MikroOrmHealthIndicator contains health indicators
+ * which are used for health checks related to MikroOrm
+ *
+ * @publicApi
+ * @module TerminusModule
+ */
+@Injectable({ scope: Scope.TRANSIENT })
+export class MikroOrmHealthIndicator extends HealthIndicator {
+  /**
+   * Initializes the MikroOrmHealthIndicator
+   *
+   * @param {ModuleRef} moduleRef The NestJS module reference
+   */
+  constructor(private moduleRef: ModuleRef) {
+    super();
+    this.checkDependantPackages();
+  }
+
+  /**
+   * Checks if responds in (default) 1000ms and
+   * returns a result object corresponding to the result
+   * @param key The key which will be used for the result object
+   * @param options The options for the ping
+   *
+   * @example
+   * MikroOrmHealthIndicator.pingCheck('database', { timeout: 1500 });
+   */
+  public async pingCheck(
+    key: string,
+    options: MikroOrmPingCheckSettings = {},
+  ): Promise<HealthIndicatorResult> {
+    this.checkDependantPackages();
+
+    const connection = options.connection || this.getContextConnection();
+    const timeout = options.timeout || 1000;
+
+    if (!connection) {
+      return this.getStatus(key, false);
+    }
+
+    try {
+      await this.pingDb(connection, timeout);
+    } catch (error) {
+      // Check if the error is a timeout error
+      if (error instanceof PromiseTimeoutError) {
+        throw new TimeoutError(
+          timeout,
+          this.getStatus(key, false, {
+            message: `timeout of ${timeout}ms exceeded`,
+          }),
+        );
+      }
+
+      // Check if the error is a connection not found error
+      if (error instanceof Error) {
+        throw new HealthCheckError(
+          error.message,
+          this.getStatus(key, false, {
+            message: error.message,
+          }),
+        );
+      }
+    }
+
+    return this.getStatus(key, true);
+  }
+
+  private checkDependantPackages() {
+    checkPackages(
+      ['@mikro-orm/nestjs', '@mikro-orm/core'],
+      this.constructor.name,
+    );
+  }
+
+  /**
+   * Returns the connection of the current DI context
+   */
+  private getContextConnection(): Connection | null {
+    const mikro = this.moduleRef.get(MikroORM, { strict: false });
+
+    const connection: Connection = mikro.em.getConnection();
+
+    if (!connection) {
+      return null;
+    }
+    return connection;
+  }
+
+  /**
+   * Pings a mikro-orm connection
+   *
+   * @param connection The connection which the ping should get executed
+   * @param timeout The timeout how long the ping should maximum take
+   *
+   */
+  private async pingDb(connection: Connection, timeout: number) {
+    let check: Promise<any>;
+    const type = connection.getPlatform().getConfig().get('type');
+
+    switch (type) {
+      case 'postgresql':
+      case 'mysql':
```

Comment:
Doesn't MariaDB & SQLite also fall into this case?

Reply:
I guess the only one missing is MongoDB then, right?

Reply:
> Doesn't MariaDB & SQLite also fall into this case?

Yeah, that should work. MongoDB is the only one missing I can implement and test if required. Left it out because I don't use it much.